### DISCUSSION TITLE
development of the default "Cassiopeia" template for the frontend

### DISCUSSION THREAD

**Discussion by 4digitalworld** [joomla/joomla-cms#43330] [2. Cold Features]
I heard that it was planned to make a template builder in @Joomla, but instead I propose to develop the default `Cassiopeia` template and add new settings and features for this template in each new minor & major version, and also continue to develop this template and keep it as the default template for the frontend, updating the @twbs **Bootstrap** version even in major updates like **Joomla 6.x**, **Joomla 7.x**, etc.

I propose to add such features for `Cassiopeia` template in the next version of Joomla (in Advanced template settings):

1. Custom `<meta>` tags in `<head>`
2. Custom code before `</head>`
3. Custom Favicon path
4. Custom code after `<body>`
5. Custom code before `</body>`
6. Custom CSS / JS code

It would be much better if these features were in the template settings rather than in the template overrides.
This would make the Joomla experience much easier for new users.

Please do not forget that Joomla will be able to receive more developers and more donations if this CMS is used by as many new users as possible who are just starting to make their first blog.

**Comment by brianteeman** [comment]
Whatever is done the core Joomla template should encourage best practice. For far too long the vast majority of the template providers have employed bad practices and we should not repeat those mistakes
